pkg/asset/installconfig/ibmcloud/validation.go

```diff
@@ -414,6 +417,54 @@ func ValidatePreExistingPublicDNS(client API, ic *types.InstallConfig, metadata
 	return nil
 }
 
+// ValidateServiceEndpoints will validate a series of service endpoint overrides.
+func ValidateServiceEndpoints(ic *types.InstallConfig) error {
+	allErrs := field.ErrorList{}
+	serviceEndpointsPath := field.NewPath("platform").Child("ibmcloud").Child("serviceEndpoints")
+	// Verify services are valid for override and are not duplicated and that are in valid URI format and accessible.
+	overriddenServices := map[string]bool{}
+	for id, service := range ic.Platform.IBMCloud.ServiceEndpoints {
+		// Check if we have a duplicate service (case is ignored)
+		if _, ok := overriddenServices[strings.ToLower(service.Name)]; ok {
+			allErrs = append(allErrs, field.Duplicate(serviceEndpointsPath.Index(id).Child("name"), service.Name))
+			continue
+		}
+		// Add service to map to track for duplicates
+		overriddenServices[strings.ToLower(service.Name)] = true
+
+		// Check that the provided service name is an acceptable override service
+		if _, ok := ibmcloud.IBMCloudServiceOverrides[strings.ToLower(service.Name)]; !ok {
+			allErrs = append(allErrs, field.Invalid(serviceEndpointsPath.Index(id).Child("name"), service.Name, "not a supported override service"))
+		}
+
+		// Check if the service URL is valid
+		err := validateEndpoint(service.URL)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(serviceEndpointsPath.Index(id).Child("url"), service.URL, err.Error()))
+			continue
+		}
+	}
+
+	// TODO(cjschaef): See about performing additional endpoint validation (all vs. subsets, regional endpoint validation, etc.)
```

Comment:
what is the status of this TODO?

Reply:
I was expecting to follow up on this later, as I wait for details from other development teams on possible configurations, environments, etc. that would determine if we wish to perform additional validation, or leave the endpoint override possibilities broader.

Reply:
I removed the TODO, as we will be leaving the configuration options more broad, due to the variety of environments/restrictions we can cover.